From f4cd809d1b77ce11663865cfb13690b0f4f6b862 Mon Sep 17 00:00:00 2001
From: Elliot Ward <elliot.ward@digital.justice.gov.uk>
Date: Mon, 28 Oct 2024 15:00:36 +0000
Subject: [PATCH] #3387094: add $context to FlySystemBridge as required for
 stream wrappers in PHP. See
 https://www.php.net/manual/en/class.streamwrapper.php#streamwrapper.props.context.
 Without this, PHP warnings are raised as the property is created dynamically.
 It would be ideal to add this to the parent class
 \Twistor\FlysystemStreamWrapper, but the PR to add there is being ignored,
 and that project has no commits since 2019.

---
 src/FlysystemBridge.php | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/FlysystemBridge.php b/src/FlysystemBridge.php
index 998c37d..1f4dbce 100644
--- a/src/FlysystemBridge.php
+++ b/src/FlysystemBridge.php
@@ -14,6 +14,13 @@ class FlysystemBridge extends FlysystemStreamWrapper implements StreamWrapperInt
 
   use StringTranslationTrait;
 
+  /**
+   * PHP-passed stream context.
+   *
+   * @var resource|null
+   */
+  public $context;
+
   /**
    * {@inheritdoc}
    */
-- 
GitLab

